### PR TITLE
test/extended/util: allow tests to be skipped based on platform

### PR DESCRIPTION
**Pull request by abhinavdahiya**
Allows certain tests to be skipped based on the platform being tested.

/cc @smarterclayton

### CODE REVIEW

File: test/extended/util/test.go [hunk 1]
```diff
@@ -431,6 +432,9 @@ var (
 
 			`Should be able to support the 1\.7 Sample API Server using the current Aggregator`, // down apiservices break other clients today https://bugzilla.redhat.com/show_bug.cgi?id=1623195
 		},
+		"[azure:Skipped]": {
```
Comment (smarterclayton) [inline test/extended/util/test.go]: I think we should flip this `[Skipped:azure]`.

File: test/extended/util/test.go [hunk 2]
```diff
@@ -200,6 +200,7 @@ func AnnotateTestSuite() {
 				break
 			}
 		}
+		excludedTestsFilter := regexp.MustCompile(strings.Join(append(excludedTests, fmt.Sprintf(`\[%s:Skipped\]`, e2e.TestContext.Provider)), `|`))
```
Comment (smarterclayton) [inline test/extended/util/test.go]: I think we should only set this if test provider platform is not empty.

Comment (abhinavdahiya) [reply test/extended/util/test.go]: +1 fixed with https://github.com/openshift/origin/compare/f3da840a3a95e884a362b837c68b6e49321fd6ac...a99a729e4239a3f3ea6c896906cd5f143f0aef96

File: test/extended/util/test.go (hunk 1 above)
Comment (abhinavdahiya) [reply test/extended/util/test.go]: +1 fixed with https://github.com/openshift/origin/compare/f3da840a3a95e884a362b837c68b6e49321fd6ac...a99a729e4239a3f3ea6c896906cd5f143f0aef96

@smarterclayton

**Comment by abhinavdahiya** [comment]
opened BZ `https://bugzilla.redhat.com/show_bug.cgi?id=1723603` for skipped storage test.

**Comment by abhinavdahiya** [comment]
ping @smarterclayton @bparees @mfojtik 

This gets us to the expectation that Azure CI should from now on be green for merge.

File: test/extended/util/test.go [hunk 3]
```diff
@@ -431,6 +435,41 @@ var (
 
 			`Should be able to support the 1\.7 Sample API Server using the current Aggregator`, // down apiservices break other clients today https://bugzilla.redhat.com/show_bug.cgi?id=1623195
 		},
+		"[Skipped:azure]": {
+			"Networking should provide Internet connection for containers", // Azure does not allow ICMP traffic to internet.
```
Comment (bparees) [inline test/extended/util/test.go]: is @openshift/sig-networking aware of this test being disabled for azure?

Comment (abhinavdahiya) [reply test/extended/util/test.go]: upstream PR for skipping azure https://github.com/kubernetes/kubernetes/pull/79243
^^-- was closed on favor of dropping ping with looks like `nc` based https://github.com/kubernetes/kubernetes/pull/77794#issuecomment-505586864

**Comment by abhinavdahiya** [comment]
ping

> ping @smarterclayton @bparees @mfojtik
> 
> This gets us to the expectation that Azure CI should from now on be green for merge.

**Pull request merged**